### PR TITLE
asset-swapper: Stop requiring takerAddress for indicative RFQ-T quotes.

### PR DESCRIPTION
**Pull request by feuGeneA**
## Description

Stop requiring takerAddress for indicative RFQ-T quotes.

## Testing instructions

Verified via automated tests over in https://github.com/0xProject/0x-api/pull/320

## Types of changes

* New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Review by PirosB3 — CHANGES_REQUESTED** [review]
When the QuoteRequestor makes a request to the RFQ client, they need to pass in a `takerAddress`. With these changes, do we risk an undefined `takerAddress`?

**Comment by feuGeneA** [comment]
> When the QuoteRequestor makes a request to the RFQ client, they need to pass in a `takerAddress`. With these changes, do we risk an undefined `takerAddress`?

That's a great question @PirosB3 , thank you for asking it.  Thinking further, I think what we need to do is modify the `@0x/quote-server` to allow indicative quotes to skip passing a `takerAddress`.  I'll put up a PR for that.

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 1]
```diff
@@ -217,7 +213,10 @@ export class QuoteRequestor {
         options: RfqtRequestOpts,
     ): Promise<RFQTIndicativeQuote[]> {
         const _opts: RfqtRequestOpts = { ...constants.DEFAULT_RFQT_REQUEST_OPTS, ...options };
-        assertTakerAddressOrThrow(_opts.takerAddress);
+
+        if (!_opts.takerAddress) {
```
Comment (PirosB3) [inline packages/asset-swapper/src/utils/quote_requestor.ts:222]: Would you mind adding a quick comment explaining why we set indicative quote `takerAddress` to `NULL_ADDRESS`?

**Review by PirosB3 — APPROVED** [review]

File: packages/asset-swapper/src/utils/quote_requestor.ts (hunk 1 above)
Comment (feuGeneA) [reply packages/asset-swapper/src/utils/quote_requestor.ts:222]: Addressed in https://github.com/0xProject/0x-monorepo/pull/2684/commits/a8f3a47240c1a0adabbdd60ca82c8c98b4f07851

**Pull request merged**